src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -0,0 +1,976 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Float = System.Single;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.TextAnalytics;
+
+[assembly: LoadableClass(typeof(LdaTransform), typeof(LdaTransform.Arguments), typeof(SignatureDataTransform),
+    LdaTransform.UserName, LdaTransform.LoaderSignature, LdaTransform.ShortName, DocName = "transform/LdaTransform.md")]
+
+[assembly: LoadableClass(typeof(LdaTransform), null, typeof(SignatureLoadDataTransform),
+    LdaTransform.UserName, LdaTransform.LoaderSignature)]
+
+/*
+ * REVIEW: So this would be my starting point. Much of the plumbing is already
+ * done, except for the parts indicated by "REVIEW taifengw(tfinley):" in comments. I have not
+ * tried this at all, so I could have some horrifying bugs and oversights here.
+ * 
+ * The major thing to be done is the stuff referring to LdaState. (This is a placeholder object.
+ * I don't suggest you name it that or anything, but it's meant to convey the idea.)
+ * 
+ * You also probably want to *not* put this here, but to instead move this transform to whatever
+ * assembly you're making. But I am not entirely certain where makes sense.
+ */
+
+namespace Microsoft.ML.Runtime.TextAnalytics
+{
+    /// <summary>
+    /// The latent Dirichlet allocation (LDA) transform. 
+    /// http://arxiv.org/abs/1412.1576 
+    /// </summary>
+    public sealed class LdaTransform : OneToOneTransformBase
+    {
+        public sealed class Arguments : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:srcs)", ShortName = "col", SortOrder = 49)]
+            public Column[] Column;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The number of topics in the LDA", SortOrder = 50)]
+            [TGUI(SuggestedSweeps = "20,40,100,200")]
+            [TlcModule.SweepableDiscreteParam("NumTopic", new object[] { 20, 40, 100, 200 })]
```

Comment:
>object [](start = 62, length = 6)

Do these really have to be `object`? `SweepableDiscreteParam` isn't a generic type?

Reply:
SweepableDiscreteParam  constructor requires that. changing to int for example breaks compilation. To have what you want will need to overload constructor with various types. if you feel strongly about it lets open separate issue to address it

---
In reply to: [197513675](https://github.com/dotnet/machinelearning/pull/377#discussion_r197513675) [](ancestors = 197513675)

Reply:
That's OK. We'll need to improve this elsewhere I suppose...

---
In reply to: [197954198](https://github.com/dotnet/machinelearning/pull/377#discussion_r197954198) [](ancestors = 197954198,197513675)